docs/load_balancer.md

```diff
@@ -0,0 +1,278 @@
+# Load balancing & publicly accessible services
+
+This document describes how kubernetes services are made accessible outside the cluster.
+
+There are two main mechanisms:
+
+* The service can be exposed through a load-balancer managed by k8s.
+* k8s can also expose services without an associated load-balancer; the service could be reached via DNS SRV records,
+or by a load-balancer configured by a helper process, or even by a client using the k8s API directly for discovery.
+_(new functionality)_
+
+## Public ports
+
+To be exposed outside the k8s cluster, the service must be on known ports on externally reachable
+IPs.  For version 1 public services, k8s listens on _every_ external IP address on every minion, and assigns
+a _single_ port cluster-wide for each port on public services.
+
+## Important scenarios
+
+The names in bold are the names we will use in the API.
+
+1. **cluster** The service is reachable from within the cluster.  _(Existing functionality)_
+1. **loadbalancer** The service is exposed via a k8s-managed load-balancer.  _(Existing functionality)_
+1. **public** The service is exposed on a set of publicly-routable IP addresses, on a set of ports.  _(New functionality)_.
```

Comment:
This spec can't actually guarantee that the IPs are routable.  All we can guarantee is that we expose it on nodes.  And even that is not quite right - will we open up firewall rules in this case?  Probably should...